src/cli/rustup_mode.rs

```diff
@@ -808,9 +811,41 @@ fn update(cfg: &Cfg, m: &ArgMatches<'_>) -> Result<()> {
         common::update_all_channels(cfg, self_update, m.is_present("force"))?;
     }
 
+    delete_downloads_dir_contents();
+
     Ok(())
 }
 
+fn delete_downloads_dir_contents() {
+    match dirs::home_dir() {
+        Some(path) => match path.to_str() {
+            Some(home_dir) => {
+                let downloads_dir_path = format!("{}/.rustup/downloads", home_dir);
```

Comment:
Paths are managed differently in Rustup because we have some environment variable overrides possible.  The `Cfg` instance which is passed to the `update()` function has a `download_dir` member which you can use.